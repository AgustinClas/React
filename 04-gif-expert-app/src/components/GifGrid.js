import React from 'react'
import { useFetchGifs } from '../hooks/useFetchGifs'
import { GifGridItem } from './GifGridItem';
import PropTypes from 'prop-types';

export const GifGrid = ({category}) => {

    const {loading, data:images} = useFetchGifs(category);

    return (
        <div className = "GifGrid">
            <h3>{category}</h3>
            {loading && <p>Cargando</p>} 
            <div className='card-grid'>
                {
                    images.map((img) =>(
                        <GifGridItem 
                        {...img}
                        key = {img.id}
                        />
                        ))
                }
            </div> 
        </div>
    )
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}