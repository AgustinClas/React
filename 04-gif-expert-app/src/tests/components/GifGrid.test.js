import React, { useState } from "react";
import {shallow} from 'enzyme';
import '@testing-library/jest-dom';
import { GifGrid } from "../../components/GifGrid";
import { useFetchGifs } from "../../hooks/useFetchGifs";
jest.mock("../../hooks/useFetchGifs");

describe('Pruebas en GifGrid', () => {

    const category = "Dragon Ball";

    let wrapper;

    /* beforeEach(() => {

        const gifs = [{id : 'abc', url: 'https://cualquiercosa.com/cosa.jpg', title: 'cualquierCosa'}];

        useFetchGifs.mockReturnValue({
            loading : false,
            data: gifs
        })

        wrapper = shallow(<GifGrid category = { category }/>);
    
    }); */
    
    test('Debe mostrar los elementos de la manera correspondiente', () => {


        useFetchGifs.mockReturnValue({
            loading : true,
            data: []
        })

        wrapper = shallow(<GifGrid category = { category }/>);

        expect( wrapper ).toMatchSnapshot();

    });

    
    test('debe mostrar items cuando se cargan imagenes useFetchGifs', () => {
        
        const gifs = [{id : 'abc', url: 'https://cualquiercosa.com/cosa.jpg', title: 'cualquierCosa'},
        {id : 'abc', url: 'https://cualquiercosa.com/cosa.jpg', title: 'cualquierCosa'}];
        
        useFetchGifs.mockReturnValue({
            loading : false,
            data: gifs
        })
        
        wrapper = shallow(<GifGrid category = { category }/>);
        
        expect( wrapper ).toMatchSnapshot();
        expect(wrapper.find('p').exists()).toBe( false );
        expect( wrapper.find('GifGridItem').length).toBe(gifs.length);

        
    });
    
    test('debe mostrar el titulo enviado como parametro', () => {
        
        const categoryAux = wrapper.find('h3').text().trim();
    
        expect(categoryAux).toBe(category);
    });
})