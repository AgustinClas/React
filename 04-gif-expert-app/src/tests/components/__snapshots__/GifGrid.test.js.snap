// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifGrid Debe mostrar los elementos de la manera correspondiente 1`] = `
<div
  className="GifGrid"
>
  <h3>
    Dragon Ball
  </h3>
  <p>
    Cargando
  </p>
  <div
    className="card-grid"
  />
</div>
`;

exports[`Pruebas en GifGrid debe mostrar items cuando se cargan imagenes useFetchGifs 1`] = `
<div
  className="GifGrid"
>
  <h3>
    Dragon Ball
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="abc"
      key="abc"
      title="cualquierCosa"
      url="https://cualquiercosa.com/cosa.jpg"
    />
    <GifGridItem
      id="abc"
      key="abc"
      title="cualquierCosa"
      url="https://cualquiercosa.com/cosa.jpg"
    />
  </div>
</div>
`;
